### ISSUE TITLE
[UX] Allows to report images

### ISSUE THREAD

**Issue by alexfauquette** [openfoodfacts/hunger-games#22]
## Technical aspect

- The interface to mark an image as reported
- A way to manage reported images

### Interface 

To avoid having too much information on the main interface, the buttons to report an image could be added in the image gallery only.
For example a flag `IconButton` in the right top corner of each image

![image](https://user-images.githubusercontent.com/45398769/175132022-54e6f567-0bf3-4608-a5ef-0661aba293e2.png)

### Managing reported images

To discuss with @teolemon :
- Where data is stored
- How do we send it
- How can we access it
- How administrators clean those problematic images

If needed I have a workaround to test the feature: 

On my personal server (amathjourney.com/api), I have an API do save flagged images:

- PUT `https://amathjourney.com/api/off-annotation/flag-image/[code]` with body containing `{imgid, url}`. The `imgid` must contain a number (the id of the image) and `url` is the image src
- GET `https://amathjourney.com/api/off-annotation/flag-image/[code]` will return all the data you put for a given `code`
- DELETE `https://amathjourney.com/api/off-annotation/flag-image/[code]` with body containing `{imgid}`

Basically, it saves all that in a database whit the primary key `code + imgid`

**Comment by teolemon** [comment]
- no strong opinion, I'd expect this to be a low volume API, hopefully, so we can do with your backend for now ?
- this api would be very helpful with the app, I'm almost tempted to use your endpoint in Smoothie 😈

**Comment by alexfauquette** [comment]
It's a small gandi shared instance I keep for a website use by 2 people per week, so you can reuse it, just be aware it's not tested, no backup

**Comment by teolemon** [comment]
Do you have the source code somewhere ?

**Comment by alexfauquette** [comment]
Not a really complex one, can be rewrite in any other language

```js
const deleteFlaggedImages = (req, res) => {
  const code = req.params.code;
  const { imgid } = req.body;

  poolOFFAnnotation
    .query("DELETE FROM flagged_images WHERE code=$1 AND imgid=$2", [
      code,
      imgid,
    ])
    .then((result) => res.status(200).json({ result: result.rows }))
    .catch((err) => res.status(500).json({ err }));
};

const addFlaggedImages = (req, res) => {
  const code = req.params.code;
  const { imgid, url } = req.body;

  poolOFFAnnotation
    .query(
      "INSERT INTO flagged_images (code, imgid, url) VALUES ($1, $2, $3)",
      [code, imgid, url]
    )
    .then((result) => res.status(200).json({ result: result.rows }))
    .catch((err) => res.status(500).json({ err }));
};

const getFlaggedImages = (req, res) => {
  res.set("Cache-control", "public, max-age=5");
  const code = req.params.code;

  poolOFFAnnotation
    .query("SELECT imgid FROM flagged_images WHERE code=$1", [code])
    .then((result) => res.status(200).json({ result: result.rows }))
    .catch((err) => res.status(500).json({ err }));
};
```

**Comment by sumana2001** [comment]
Where will all this data be stored? How can we access that database?

**Comment by alexfauquette** [comment]
@sumana2001 You can write the code with the following API. If contributors get used to the interface, no doubt that we will get either this behavior added to robotoff or a dedicated server later on.

> On my personal server (amathjourney.com/api), I have an API do save flagged images:
> 
> PUT https://amathjourney.com/api/off-annotation/flag-image/[code] with body containing {imgid, url}. The imgid must contain a number (the id of the image) and url is the image src
> GET https://amathjourney.com/api/off-annotation/flag-image/[code] will return all the data you put for a given code
> DELETE https://amathjourney.com/api/off-annotation/flag-image/[code] with body containing {imgid}
